packages/gradle/project-graph/src/main/kotlin/dev/nx/gradle/utils/TaskUtils.kt

```diff
@@ -0,0 +1,354 @@
+              externalDep?.let { externalDependencies.add(it) }
+            } catch (e: Exception) {
+              task.logger.info("${task}: get external dependency error $e")
+            }
+          }
+        }
+    if (externalDependencies.isNotEmpty()) {
+      mappedInputsIncludeExternal.add(mutableMapOf("externalDependencies" to externalDependencies))
+    }
+    if (mappedInputsIncludeExternal.isNotEmpty()) {
+      return mappedInputsIncludeExternal
+    }
+    return null
+  } catch (e: Exception) {
+    // Log the error but don't fail the build
+    task.logger.info("Error getting outputs for ${task.path}: ${e.message}")
+    task.logger.debug("Stack trace:", e)
+    null
+  }
+}
+
+/**
+ * Get outputs for task
+ *
+ * @param task task to process
+ * @return list of outputs file, will not include if output file is outside workspace, null if empty
+ *   or an error occurred
+ */
+fun getOutputsForTask(task: Task, projectRoot: String, workspaceRoot: String): List<String>? {
+  return try {
+    val outputs = task.outputs.files
+    if (!outputs.isEmpty) {
+      return outputs.mapNotNull { file ->
+        val path: String = file.path
+        replaceRootInPath(path, projectRoot, workspaceRoot)
+      }
+    }
+    null
+  } catch (e: Exception) {
+    // Log the error but don't fail the build
+    task.logger.info("Error getting outputs for ${task.path}: ${e.message}")
+    task.logger.debug("Stack trace:", e)
+    null
+  }
+}
+
+/**
+ * Get dependsOn for task, handling configuration timing safely. Rewrites dependency task names
+ * based on targetNameOverrides (e.g., test -> ci).
+ *
+ * @param task task to process
+ * @param dependencies optional set to collect inter-project Dependency objects
+ * @param targetNameOverrides optional map of overrides (e.g., test -> ci)
+ * @return list of dependsOn task names (possibly replaced), or null if none found or error occurred
+ */
+fun getDependsOnForTask(
+    task: Task,
+    dependencies: MutableSet<Dependency>?,
+    targetNameOverrides: Map<String, String> = emptyMap()
+): List<String>? {
+
+  fun mapTasksToNames(tasks: Collection<Task>): List<String> {
+    return tasks.map { depTask ->
+      val depProject = depTask.project
+      val taskProject = task.project
+
+      if (task.name != "buildDependents" && depProject != taskProject && dependencies != null) {
+        dependencies.add(
+            Dependency(
+                taskProject.projectDir.path,
+                depProject.projectDir.path,
+                taskProject.buildFile.path))
+      }
+
+      // Check if this task name needs to be overridden
+      var taskName = targetNameOverrides.getOrDefault(depTask.name + "TargetName", depTask.name)
+      val ciTargetName = targetNameOverrides.getOrDefault("ciTargetName", null)
+      if (depTask.name === "test" && ciTargetName != null && System.getenv("CI").equals("true")) {
+        taskName = ciTargetName
+      }
```

Comment:
We shouldn't have different targets for CI and non-CI.

Let's keep `build` depending on `test` as is.

Let's instruct Gradle users to run `nx affected -t assemble,check-ci` which will not run `test`.. but `test-ci`.

Reply:
changed